### PR TITLE
feat: Add FormLayout component

### PR DESCRIPTION
**Pull request by diegonzs-roll**
## What's done
- Create FormLayout component
## Tasks

- [ ] Have you written tests for new code (if applicable)?
- [x] Have you tested the changes on all the platforms?
  - [x] iOS
  - [ ] Android
  - [x] Web
- [x] Have you added stories to demonstrate the new functionality (if there is any)?

## Demo (screenshots, recordings, etc.)
https://www.loom.com/share/5f875e76e28245479db6f8769f873a69

### CODE REVIEW

File: packages/design-system/src/organisms/formStepHeader/index.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      showsHorizontalScrollIndicator={false}
+      data={steps}
+      ItemSeparatorComponent={Separator}
+      keyExtractor={(item) => item.title}
+      onScrollToIndexFailed={(info) => {
+        const wait = new Promise((resolve) => setTimeout(resolve, 500))
+        wait.then(() => {
+          listRef.current?.scrollToIndex({ index: info.index, animated: true })
+        })
+      }}
```
Comment (teimurjan) [inline packages/design-system/src/organisms/formStepHeader/index.tsx:107]: This retry might lead to an endless loop. Let's extract it to a function (outside of the component):

```ts
// TODO: replace any
const SCROLL_RETRY_TIMEOUT = 500

const makeScrollToIndexFailedHandler = (listRef: any) => {
  let attemptsLeft = 3
  let timeoutId: number | undefined = undefined
  // TODO: replace any
  return (info: any) => {
    if (timeoutId) {
      clearTimeout(timeoutId)
    }
    if (attemptsLeft === 0) {
      return
    }
    timeoutId = setTimeout(() => {
      listRef.current?.scrollToIndex({ index: info.index, animated: true })
      timeoutId = undefined
      attemptsLeft--
    }, SCROLL_RETRY_TIMEOUT)
  }
}
```

Then, before the component's return statement:

```ts
const onScrollToIndexFailed = useMemo(() => makeScrollToIndexFailedHandler(listRef), [listRef])
```

And, pass to the props:

```suggestion
      onScrollToIndexFailed={onScrollToIndexFailed}
```

Comment (teimurjan) [reply packages/design-system/src/organisms/formStepHeader/index.tsx:107]: You can extract the function to `utils` file & add a test for it.

File: packages/design-system/src/organisms/formStepHeader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      renderItem={({ item, index }) => (
+        <Step
+          step={index + 1}
```
Comment (teimurjan) [inline packages/design-system/src/organisms/formStepHeader/index.tsx:110]: If we add `id` to the props, we'll be able to use it here.

File: packages/design-system/src/organisms/formStepHeader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          container.center,
+          margin.mr16,
+          { backgroundColor, borderColor },
+        ]}
+      >
+        <TypographyV2
+          variant="caption1"
+          color={textColor}
+          style={[
+            Platform.select({ native: { lineHeight: NATIVE_LINE_HEIGHT } }),
```
Comment (teimurjan) [inline packages/design-system/src/organisms/formStepHeader/index.tsx:68]:
```suggestion
            Platform.select({ native: { lineHeight: NATIVE_LINE_HEIGHT }, default: undefined }),
```

File: packages/design-system/src/organisms/formStepHeader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        ]}
+      />
+    </View>
+  )
+}
+
+interface StepProps {
+  step: number
+  title: string
+  isChecked: boolean
```
Comment (teimurjan) [inline packages/design-system/src/organisms/formStepHeader/index.tsx:46]: Let's follow the same naming convention as `disabled` & `editable` here:

```suggestion
  checked: boolean
```

File: packages/design-system/src/utils/functions.ts [hunk 5]
```diff
@@ -14,3 +14,24 @@ export const openLink = (link: string, newTab = false) => {
 
 export const isLast = (index: number, array: unknown[]) =>
   index === array.length - 1
+
+const SCROLL_RETRY_TIMEOUT = 500
+export const makeScrollToIndexFailedHandler = (
```
Comment (teimurjan) [inline packages/design-system/src/utils/functions.ts:19]: I think the function should not be here. It's not global-scoped. Let's move it under `formStepHeader` folder in a filed named `utils.ts`. Also, I'd like to have this function unit tested (it's a valuable logic in my opinion :))

File: packages/design-system/src/organisms/formStepHeader/index.tsx (hunk 3 above)
Comment (diegonzs-roll) [reply packages/design-system/src/organisms/formStepHeader/index.tsx:68]: I'll use `web` instead of `default` because is not an accepted value.

**Review by teimurjan — APPROVED** [review]

**Pull request merged**